### PR TITLE
AV-177986 : Support for OVN-Kubernetes CNI in Openshift with AKO

### PR DESCRIPTION
**Pull request by DixitAakash**
This PR is for adding support for OVN-Kubernetes CNI in Openshift with AKO.

### CODE REVIEW

**Comment by DixitAakash** [comment]
build ako

File: internal/lib/dynamic_client.go [hunk 1]
```diff
@@ -274,6 +275,24 @@ func GetPodCIDR(node *v1.Node) ([]string, error) {
 			}
 		}
 
+	} else if GetCNIPlugin() == OVN_KUBERNETES_CNI {
+		subnetAnnotation := "k8s.ovn.org/node-subnets"
```
[the remark this answers is not in the document]
Comment (DixitAakash) [reply internal/lib/dynamic_client.go]: Done

File: internal/lib/dynamic_client.go [hunk 2]
```diff
@@ -274,6 +275,24 @@ func GetPodCIDR(node *v1.Node) ([]string, error) {
...
+		var nodeSubnets string
+		var found bool
+		if nodeSubnets, found = node.Annotations[subnetAnnotation]; !found {
+			return nil, errors.New("k8s.ovn.org/node-subnets annotation not found in Node Metadata")
+		}
+		var nodeSubnetJson map[string]interface{}
```
[the remark this answers is not in the document]
Comment (DixitAakash) [reply internal/lib/dynamic_client.go:285]: Add UT's for this change.

**Review by swathinsankaran — APPROVED** [review]
LGTM

**Comment by swathinsankaran** [comment]
build ako

File: internal/lib/dynamic_client.go [hunk 3]
```diff
@@ -274,6 +275,24 @@ func GetPodCIDR(node *v1.Node) ([]string, error) {
 			}
 		}
 
+	} else if GetCNIPlugin() == OVN_KUBERNETES_CNI {
+		subnetAnnotation := OVNNodeSubnetAnnotation
+		var nodeSubnets string
+		var found bool
+		if nodeSubnets, found = node.Annotations[subnetAnnotation]; !found {
```
[the remark this answers is not in the document]
Comment (DixitAakash) [reply internal/lib/dynamic_client.go]: Done

File: internal/lib/dynamic_client.go [hunk 4]
```diff
@@ -274,6 +275,24 @@ func GetPodCIDR(node *v1.Node) ([]string, error) {
 			}
 		}
 
+	} else if GetCNIPlugin() == OVN_KUBERNETES_CNI {
```
[the remark this answers is not in the document]
Comment (DixitAakash) [reply internal/lib/dynamic_client.go:278]: Is IPV6 supported for pod cidrs in Openshift nodes? Also, I am not sure about configuring multiple pod cidrs for a node. Currently, getiing the pod cidr from this annotation `k8s.ovn.org/node-subnets: '{"default":"10.128.2.0/23"}'` and populating the static routes. Hence, I have added the **testing-in-progress** and **do-not-merge** labels to this PR.

**Comment by DixitAakash** [comment]
build ako

**Review by swathinsankaran — APPROVED** [review]

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**